### PR TITLE
Use git URL for normal cargo build, and vendored file path for Nix build

### PR DESCRIPTION
**Pull request by soareschen**
## Summary

- Removes the use of git submodule.
- Use normal Git URL to fetch the private sovereign-sdk repository when building with `cargo build`.
- Use vendored file path when building in Nix.
- Use separate toml entries for each Sovereign SDK dependency
  - Each entry would have uncommented git URL, and commented path value
  - The Nix build would uncomment the path value, and comment out the git URLs.

## Motivation

The use of git submodule turns out to not work well with Cargo. When one tries to run `cargo build` with git dependencies with git submodules, Cargo will almost always try to also fetch the git submodule, whether it is required in the Cargo dependency or not.

Because of this, we need to use back git dependency for Sovereign SDK modules. On the other hand, in Nix we would patch the git dependencies, and replace them with path dependencies in order to work in Nix. This is because the Nix Rust build still does not support private repository, and we still need to use local vendored file path dependency inside the Nix build.

### CODE REVIEW

File: nix/sov-celestia-cw.nix [hunk 1]
```diff
@@ -16,11 +52,11 @@ let
...
             cp -r ${../mocks} $out/mocks
             cp -r ${../proto} $out/proto
+
             cp -r ${sovereign-sdk-src} $out/vendor/sovereign-sdk
-            cp ${../Cargo.lock} $out/Cargo.lock
 
-            cat ${../Cargo.toml} > $out/Cargo.toml
-            cat ${../.cargo/config.toml} >> $out/Cargo.toml
+            cp ${cargo-lock-file} $out/Cargo.lock
+            cat ${cargo-toml-file} > $out/Cargo.toml
```
Comment (rnbguy) [inline nix/sov-celestia-cw.nix:59]: Is this to replace the exisiting `$out/Cargo.toml`?

Comment (soareschen) [reply nix/sov-celestia-cw.nix:59]: In the new code, we would do string manipulation in Nix to replace the git URLs with file paths in Cargo.toml. We then write the output to the intermediary source code, which has the Cargo.toml replaced with the modified version.

Comment (rnbguy) [reply nix/sov-celestia-cw.nix:59]: Sounds good. is there any reason for using `cat` and `cp` together?

Comment (soareschen) [reply nix/sov-celestia-cw.nix]: Ah, that was an artifact from the previous edit. Nice catch, fixed now.

**Review by rnbguy — APPROVED** [review]
LGTM :+1: Thanks Soares :pray:

**Pull request merged**